cmake/oneDPLWindowsIntelLLVMConfig.cmake

```diff
@@ -0,0 +1,105 @@
+##===-- oneDPLWindowsIntelLLVMConfig.cmake --------------------------------------===##
+#
+# Copyright (C) Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# This file incorporates work covered by the following copyright and permission
+# notice:
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+#
+##===----------------------------------------------------------------------===##
+cmake_minimum_required(VERSION 3.15.0)
+
+include(${CMAKE_ROOT}/Modules/FindPackageHandleStandardArgs.cmake)
+
+#if this is included after project() and oneDPLWindowsIntelLLVM hasn't already been included (in a previous cmake call), we are too late to make the compiler changes we need to
+if (DEFINED CMAKE_PROJECT_NAME AND (NOT DEFINED oneDPLWindowsIntelLLVM_DIR))
+    set(oneDPLWindowsIntelLLVM_FOUND False)
+    set(REASON_FAILURE "oneDPLWindowsIntelLLVM package must be included before the project() call!")
+else()
+
+    # CMAKE_CXX_COMPILER and CMAKE_CXX_COMPILER_VERSION cannot be used because
+    # CMake 3.19 and older will detect IntelLLVM compiler as CLang with CLang-specific version, see https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
+    if (CMAKE_CXX_COMPILER MATCHES ".*(dpcpp-cl|dpcpp|icx-cl|icpx|icx)(.exe)?$")
+        set(INTEL_LLVM_COMPILER TRUE)
+        execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_VARIABLE INTEL_LLVM_COMPILER_VERSION_RAW)
+        string(REGEX MATCH "[0-9][0-9][0-9][0-9]\\.[0-9]\\.[0-9]" INTEL_LLVM_COMPILER_VERSION ${INTEL_LLVM_COMPILER_VERSION_RAW})
```

Comment:
It seems like this regex is matching versions with 3 components, but it is used later to compare to versions like `2021.4` and `2023.1`, which have 2 components. Is that okay?